osu.Game.Tests/Visual/Menus/TestSceneExitingScreens.cs

```diff
@@ -0,0 +1,151 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Linq;
+using NUnit.Framework;
+using osu.Framework.Allocation;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Shapes;
+using osu.Framework.Platform;
+using osu.Framework.Screens;
+using osu.Framework.Testing;
+using osu.Game.Graphics.UserInterface;
+using osu.Game.Overlays.Mods;
+using osu.Game.Screens.Menu;
+using osu.Game.Screens.Select;
+using osuTK.Graphics;
+using osuTK.Input;
+
+namespace osu.Game.Tests.Visual.Menus
+{
+    public class TestSceneExitingScreens : ManualInputManagerTestScene
+    {
+        private readonly TestOsuGame osuGame = new TestOsuGame();
+
+        [BackgroundDependencyLoader]
+        private void load(GameHost gameHost)
+        {
+            osuGame.SetHost(gameHost);
+
+            Children = new Drawable[]
+            {
+                new Box
+                {
+                    RelativeSizeAxes = Axes.Both,
+                    Colour = Color4.Black,
+                },
+                osuGame
+            };
+        }
+
+        [SetUpSteps]
+        public void SetUpSteps()
+        {
+            AddUntilStep("wait for load", () => osuGame.IsLoaded);
+            AddUntilStep("exit to main menu", () =>
```

Comment:
what's the purpose of this step? how would we be anywhere else?

Reply:
We could potentially still be in the intro or disclaimer, and this makes sure you can reset a test in visual test browser even if you stop in the middle of one.

Reply:
that's not how it should be done. "resetting" should be making a new game. you're just asking for this to fail in random ways.

Reply:
I guess we should create a game instance that skips directly to the main menu on run to avoid this step entirely.

Reply:
the step is fine. it shouldn't be calling exit